service/s3/s3crypto/s3crypto_encryption_client.go

```diff
@@ -115,20 +137,23 @@ func (c *EncryptionClient) PutObjectRequest(input *s3.PutObjectInput) (*request.
 		shaHex := hex.EncodeToString(sha.GetValue())
 		req.HTTPRequest.Header.Set("X-Amz-Content-Sha256", shaHex)
 
-		f.Seek(0, 0)
-		input.Body = f
+		dst.Seek(0, 0)
+		input.Body = dst
 
 		err = c.Config.SaveStrategy.Save(env, input)
 		r.Error = err
 	})
 
-	fn := func(r *request.Request) {
-		// Close the temp file and cleanup
-		f.Close()
-		os.Remove(f.Name())
+	if useTempFile {
+		fn := func(r *request.Request) {
+			f := dst.(*os.File)
+			// Close the temp file and cleanup
+			f.Close()
+			os.Remove(f.Name())
+		}
+		req.Handlers.Send.PushBack(fn)
+		req.Handlers.ValidateResponse.PushBack(fn)
```

Comment:
Don't need to attempt to clean up twice. I suggest just do it on Send. Doing cleanup on both could be source of errors in future.